File: ischia/src/main/java/cz/fb/manaus/spring/IschiaLocalConfiguration.java
package cz.fb.manaus.spring;

import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.FilterType;
import org.springframework.context.annotation.Import;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Repository;

@Configuration
@Profile("ischia")
@ComponentScan(value = "cz.fb.manaus.ischia",
        excludeFilters = {
                @ComponentScan.Filter(type = FilterType.ANNOTATION, value = Repository.class),
                @ComponentScan.Filter(type = FilterType.ANNOTATION, value = DatabaseComponent.class),
        })
@Import({BetfairValues.class, MatchbookValues.class})
public class IschiaLocalConfiguration {

}

File: ischia/src/main/java/cz/fb/manaus/spring/IschiaLocalConfiguration.kt
package cz.fb.manaus.spring

import cz.fb.manaus.core.model.TYPE_MATCH_ODDS
import cz.fb.manaus.reactor.betting.listener.MarketRunnerPredicate
import cz.fb.manaus.spring.conf.MarketRunnerConf
import cz.fb.manaus.spring.conf.PriceConf
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.*


@Configuration
@Profile("ischia")
@ComponentScan(value = ["cz.fb.manaus.ischia"])
@Import(ValidationConfiguration::class, BettorConfiguration::class, ProfitMetricConfiguration::class)
@EnableConfigurationProperties(MarketRunnerConf::class, PriceConf::class)
open class IschiaLocalConfiguration {

    @Bean
    open fun marketRunnerPredicate(marketRunnerConf: MarketRunnerConf): MarketRunnerPredicate {
        val runnerTest = marketRunnerConf.runnerName ?: ""
        return { market, runner ->
            when (market.type ?: error("missing type")) {
                TYPE_MATCH_ODDS -> runnerTest in runner.name.toLowerCase()
                else -> true
            }
        }
    }
}